rascaline-c-api/src/lib.rs

```diff
@@ -8,7 +8,56 @@ mod utils;
 #[macro_use]
 mod status;
 pub use self::status::{catch_unwind, rascal_status_t};
+pub use log::{error, info, warn, Record, Level, Metadata, LevelFilter};
+use std::ffi::{CString};
+
 
 mod system;
 mod descriptor;
 mod calculator;
+
+pub type RascalLoggingCallback = Option<unsafe extern fn(level: i32, message: *const std::os::raw::c_char)>;
+
+
+static mut GLOBAL_CALLBACK: RascalLoggingCallback = None;
+struct RascalLogger;
+
+#[no_mangle]
+pub unsafe extern fn rascal_set_logging_callback(callback: RascalLoggingCallback) {
+    GLOBAL_CALLBACK = callback;
```

Comment:
This needs to go through a Mutex to prevent data races if two threads want to set callback at the same time